liquidhaskell-boot/src/Language/Haskell/Liquid/Bare/Misc.hs

```diff
@@ -153,9 +154,13 @@ mapTyVars _ hsT lqT
   = do err <- gets errmsg
        throwError (err (F.pprint hsT) (F.pprint lqT))
 
+-- Niki believes there is a bug on the isTYPEorCONSTRAINT function in GHC-9.12.2
+-- Hardcoding "GHC.Types.Type" for now because typeKind in in the Prim module... 
 isKind :: Kind -> Bool
-isKind = isTYPEorCONSTRAINT -- TODO:GHC-863 isStarKind k --  typeKind k
-
+isKind k = isTYPEorCONSTRAINT k -- TODO:GHC-863 isStarKind k --  typeKind k
+          || case k of 
+                TyVarTy kk -> showPpr (varType kk) == ghcTypeStr
```

Comment:
It would be better not to hardcode the string, as GHC sometimes moves definitions and the string changes, requiring debugging to catch it.

`varType kk == GHC.Builtin.Types.liftedTypeKind` would be cheaper to maintain.